React/Modules/RCTTiming.m

```diff
@@ -171,28 +171,35 @@ - (dispatch_queue_t)methodQueue
   return RCTJSThread;
 }
 
+// Called from JS Thread.
 - (void)invalidate
 {
   [self stopTimers];
   _bridge = nil;
 }
 
+// Called from main thread
 - (void)appDidMoveToBackground
 {
-  // Deactivate the CADisplayLink while in the background.
-  [self stopTimers];
-  _inBackground = YES;
-
-  // Issue one final timer callback, which will schedule a
-  // background NSTimer, if needed.
-  [self didUpdateFrame:nil];
+  [_bridge dispatchBlock:^{
+    // Deactivate the CADisplayLink while in the background.
+    [self stopTimers];
+    self->_inBackground = YES;
+    
+    // Issue one final timer callback, which will schedule a
+    // background NSTimer, if needed.
+    [self didUpdateFrame:nil];
+  } queue:RCTJSThread];
 }
 
+// Called from main thread
```

Comment:
Could you please use `RCTAssertMainQueue()` (and co) instead of comments?

Reply:
@shergin 🤔 Emm, actually, these two method can be called on any threads, because we always dispatch work to JS Thread via `dispatchBlock:queue`. Should we need to add it?

Reply:
I removed the comments. We allow it to be called on any thread. :)